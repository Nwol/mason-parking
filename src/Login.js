import React,{Component} from 'react';
import LoginController from './LoginController';
import LoginViewModel from './LoginViewModel';
import LoginStore from './models/LoginStore';


class Login extends Component {
    constructor(props){
        super(props)
        const loginModel = props[LoginStore.type.LOGIN_MODEL];
        this.viewModel = new LoginViewModel(loginModel);
    }
    render() {
        return (
            <LoginController viewModel={this.viewModel}/>
        )
    }
}

export default Login;

