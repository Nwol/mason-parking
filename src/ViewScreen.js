import React, { Component } from 'react';
import { Map, GoogleApiWrapper, Marker, InfoWindow, Polyline } from 'google-maps-react';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';

export class ViewScreen extends Component {
	constructor(props) {
		super(props);
		this.state = { showingInfoWindow: false, activeMarker: {}, selectedPlace: {}, cardType: '', polyLineVal: [] };
		this.mapStyles = { width: '100%', height: '100%' };
		this.parkingLots = [
			{ lat: 38.826775, lng: -77.307994, name: 'lotA' },
			{ lat: 38.825438, lng: -77.305887, name: 'lotC' },
			{ lat: 38.831276, lng: -77.300749, name: 'lotD' },
			{ lat: 38.832948, lng: -77.311446, name: 'lotI' },
			{ lat: 38.828801, lng: -77.312064, name: 'lotJ' },
			{ lat: 38.828259, lng: -77.311811, name: 'lotK' }
		];
		this.findParking.bind(this);
		this.onMarkerClick.bind(this);
		this.onClose.bind(this);
	}

	onMarkerClick = (props, marker, e) =>
		this.setState({
			selectedPlace: props,
			activeMarker: marker,
			showingInfoWindow: true
		});

	onClose = props => {
		if (this.state.showingInfoWindow) {
			this.setState({
				showingInfoWindow: false,
				activeMarker: null
			});
		}
	};

	findParking = (e) => {
		e.preventDefault();
		let chosenVal = {};
		let smallLatVal = 0;
		let smallLongVal = 0;
		let stateSetterVal = [];
		for (let i = 0; i < this.parkingLots.length; i++) {
			if (i === 0) {
				smallLatVal = this.parkingLots[i]['lat'] - 38.831500;
				smallLongVal = this.parkingLots[i]['lng'] - (-77.311488);
				chosenVal = this.parkingLots[i];
				continue;
			}
			else {
				if (this.parkingLots[i]['lat'] - 38.831500 < smallLatVal && this.parkingLots[i]['lng'] - (-77.311488) < smallLongVal) {
					smallLatVal = this.parkingLots[i]['lat'] - 38.831500;
					smallLongVal = this.parkingLots[i]['lng'] - (-77.311488);
					chosenVal = this.parkingLots[i];
				}
				else {
					continue;
				}
			}
		}

		stateSetterVal.push(
			<Polyline
				options={{
					path: [{ lat: 38.831500, lng: -77.311488 }, { lat: chosenVal['lat'], lng: chosenVal['lng'] }],
					strokeColor: '#FF0000',
					strokeOpacity: 0.8,
					strokeWeight: 4
				}} />
		);
		this.setState({polyLineVal: stateSetterVal});
		console.log(this.state.polyLineVal);
	};

	render() {

		let markerArray = [];
		this.parkingLots.map(data => {
			return markerArray.push(<Marker
				key={data['name']}
				onClick={this.onMarkerClick}
				name={data['name']}
				position={{ lat: data['lat'], lng: data['lng'] }}
			/>);
		});
		return (
			<div className="row no-gutters">
				{this.props.user && (<Card bg="info" text="white" style={{ width: '18rem' }}>
					<Card.Header>Where TF did I park?</Card.Header>
					<Card.Body>
						<Card.Title>Click the button below to get directions to where you parked.</Card.Title>
							<Button onClick={this.findParking} variant="success">Find my parking spot</Button>
					</Card.Body>

				</Card>)}
				<div className="col order-last map">
					<Map
						google={this.props.google}
						zoom={15.5}
						style={this.mapStyles}
						initialCenter={{
							lat: 38.831534,
							lng: -77.311587
						}}
					>
						<Marker
							onClick={this.onMarkerClick}
							name={'George Mason University'}
						/>
						{markerArray}
						{this.state.polyLineVal}
						<InfoWindow
							marker={this.state.activeMarker}
							visible={this.state.showingInfoWindow}
							onClose={this.onClose}
						>
							<div>
								<h4>{this.state.selectedPlace.name}</h4>
							</div>
						</InfoWindow>
					</Map>
				</div>
			</div>
		);
	}
}

export default GoogleApiWrapper({
	apiKey: 'AIzaSyBlKEmXM6RHfCq2eYdxUtTMfbu-XTCqVIw'
})(ViewScreen);