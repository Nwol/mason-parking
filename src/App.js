import React, { Component } from 'react';
import './App.css';

import {HashRouter, Redirect, Route} from "react-router-dom";
import Login from "./Login"
import firebase from "./firebase"
import {Nav, Navbar, NavItem} from "react-bootstrap";
import {LinkContainer} from "react-router-bootstrap"

class App extends Component {
  constructor(props){
    super(props);
    this.state={user: ''};
  }

  componentWillMount() {
    var _this = this;
    console.log(firebase);
    firebase.auth.onAuthStateChanged(function (user) {
      _this.setState({user: ''});
    }, function(error) {
      console.log(error);
    });
  }

  /**
   * Add other navs later
   */
  render() {
    let nav;
    if(this.state.user){
      nav = <Nav>
        <LinkContainer to={"/whereDidIPark"}>
          <NavItem>Where Did I Park?</NavItem>
        </LinkContainer>
        <LinkContainer to={"/viewScreen"}>
          <NavItem>View Parking Map</NavItem>
        </LinkContainer>
        <LinkContainer to={"/logout"}>
          <NavItem>Logout</NavItem>
        </LinkContainer>
      </Nav>
    }
    else{
      nav = <Nav>
          <LinkContainer to={"/login"}>
            <NavItem>Register/Sign-in</NavItem>
          </LinkContainer>
      </Nav>
    }
    if(this.state.user === undefined){
      return <span>Loading...</span>
    }

    return (
      <HashRouter>
        <div className="App">
          <Navbar>
            <Navbar.Header>
              <LinkContainer to={"/"}>
                <Navbar.Brand>Mason Parking</Navbar.Brand>
              </LinkContainer>
            </Navbar.Header>
            {nav}
          </Navbar>
          <div className={"content"}>
            <div className={"container"}>
              <Route exact path={"/"}/>
              <Route exact path={"/login"} component={Login}/>
            </div>
          </div>
        </div>
      </HashRouter>
    );
  }

  logout(){
    firebase.auth.signOut();
    return <Redirect to="/"/>
  }


  
}

export default App;
