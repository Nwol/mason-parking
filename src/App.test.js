import React from 'react';
import ReactDOM from 'react-dom';
import { create } from "react-test-renderer"; 
import App from './App';
import ViewScreen from './ViewScreen';
import Login from './Login';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('view screen', () => {
  test('matches the snapshot', () => {
    const component = create (
      <ViewScreen/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('authenticated viewscreen', () => {
  test('matches the snapshot', () => {
    const component = create (
      <ViewScreen user={true}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('login screen', () => {
  test('matches the snapshot', () => {
    const component = create (
      <Login/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
