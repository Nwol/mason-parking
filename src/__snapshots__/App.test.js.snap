// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authenticated viewscreen matches the snapshot 1`] = `
<div>
  Loading...
</div>
`;

exports[`login screen matches the snapshot 1`] = `
<div
  className="login container"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        Mason Email:
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        name="email"
        onChange={[Function]}
        placeholder="Enter email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Log in(or register)
    </button>
  </form>
</div>
`;

exports[`view screen matches the snapshot 1`] = `
<div>
  Loading...
</div>
`;
